From 4e132f13e81a8a76c6a7b36e7bbf0452323d7d80 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 16 May 2019 21:31:55 +0200
Subject: [RFC/PATCH 3/4] fsck: refactor fsck_ident() for a follow-up change
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Refactor the ident checking code added in d4b8de0420 ("fsck: report
integer overflow in author timestamps", 2014-02-24) to make a
subsequent change smaller. This also has the added benefit of making
it easier to step through this code in a debugger while examining
intermediate variables.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 fsck.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fsck.c b/fsck.c
index 4703f55561..45f82b2d40 100644
--- a/fsck.c
+++ b/fsck.c
@@ -730,6 +730,7 @@ static int fsck_ident(const char **ident, struct object *obj, struct fsck_option
 {
 	const char *p = *ident;
 	char *end;
+	timestamp_t date;
 
 	*ident = strchrnul(*ident, '\n');
 	if (**ident == '\n')
@@ -754,7 +755,8 @@ static int fsck_ident(const char **ident, struct object *obj, struct fsck_option
 	p++;
 	if (*p == '0' && p[1] != ' ')
 		return report(options, obj, FSCK_MSG_ZERO_PADDED_DATE, "invalid author/committer line - zero-padded date");
-	if (date_overflows(parse_timestamp(p, &end, 10)))
+	date = parse_timestamp(p, &end, 10);
+	if (date_overflows(date))
 		return report(options, obj, FSCK_MSG_BAD_DATE_OVERFLOW, "invalid author/committer line - date causes integer overflow");
 	if ((end == p || *end != ' '))
 		return report(options, obj, FSCK_MSG_BAD_DATE, "invalid author/committer line - bad date");
-- 
2.21.0.1020.gf2820cf01a

